### PR TITLE
feat(blueboxes): set up executor and page structure (DEV-1078)

### PR DESCRIPTION
**Pull request by EricSommerhalder**
resolves DEV-1078

### CODE REVIEW

File: src/BlueBoxes/Executor.elm [hunk 1]
```diff
@@ -1,159 +1,49 @@
...
+            let
+                ( model, cmd ) =
+                    Api.initProjectDescription wid
+            in
+            ( { variant = GuiElement.ProjectDescriptionElement model }, Cmd.map GuiElement.ProjectDescriptionMsg cmd )
+
+        Struct.Accordion wid ->
+            let
+                ( model, cmd ) =
+                    Api.initAccordion wid
```
Comment (domsteinbach) [inline src/BlueBoxes/Executor.elm:47]: "wid"

File: src/Modules/Text/Accordion.elm [hunk 2]
```diff
@@ -3,108 +3,186 @@ module Modules.Text.Accordion exposing (..)
...
-            if model.isOpen then
-                Dtw.classList
-                    [ Dtw.rounded_lg
-                    , Dtw.border_2
-                    , Dtw.cursor_pointer
-                    , Dtw.custom_bg Secondary
-                    , Dtw.custom_border Secondary
-                    ]
+init : WidgetInstanceId -> ( Model, Cmd Msg )
+init wid =
```
Comment (domsteinbach) [inline src/Modules/Text/Accordion.elm:36]: "wid" maybe better "wId"

File: mock-api/guiElements.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        },
+    "accordion": { 
+        "text": "accordion text",
+        "size": "Half"
```
Comment (domsteinbach) [inline mock-api/guiElements.json:9]: If there will be two variants/types of sizes possible in the future, a bool would be more appropriate, e.g. "fullsize: True"

File: src/Projects/TestPage.elm [hunk 4]
```diff
@@ -0,0 +1,36 @@
+module Projects.TestPage exposing (..)
+
```
Comment (domsteinbach) [inline src/Projects/TestPage.elm:2]: Testpage should be git removed

**Review by domsteinbach — CHANGES_REQUESTED** [review]
Very happy with the PR. There are two minor changes we request: Remove Testpage.elm and rename "wid" into "wId". Thank you!

File: src/BlueBoxes/Executor.elm [hunk 5]
```diff
@@ -1,159 +1,49 @@
...
-
-updateFooterPart : Msg -> Struct.FooterPart -> Struct.FooterPart
-updateFooterPart msg footer =
-    case footer of
-        Struct.AppFooter footerModel ->
-            case msg of
-                -- this currently doesn't do anything because nothing in the footer has an update function
-                _ ->
-                    footer
+        Struct.ProjectDescription wid ->
```
[the remark this answers is not in the document]
Comment (mdelez) [reply src/BlueBoxes/Executor.elm:37]: I feel like all of those have the same issue. Renaming to "widgetInstanceId" or "widgetId" would be the best.

File: mock-api/guiElements.json (hunk 3 above)
Comment (mdelez) [reply mock-api/guiElements.json:9]: A bool here makes sense actually. For the type variant names, should we use "is" in front of the names for bools? ex: "IsFullWidth"

File: src/Projects/TestPage.elm (hunk 4 above)
Comment (mdelez) [reply src/Projects/TestPage.elm]: done in [d3a4f50](https://github.com/dasch-swiss/dsp-cpe-app/pull/66/commits/d3a4f506b310a83bd7463c7d3c58663646327058)

File: src/BlueBoxes/Executor.elm (hunk 5 above)
[the remark this answers is not in the document]
Comment (mdelez) [reply src/BlueBoxes/Executor.elm]: done in [d3a4f50](https://github.com/dasch-swiss/dsp-cpe-app/pull/66/commits/d3a4f506b310a83bd7463c7d3c58663646327058)

File: mock-api/guiElements.json (hunk 3 above)
Comment (mdelez) [reply mock-api/guiElements.json]: done in [db32bf6](https://github.com/dasch-swiss/dsp-cpe-app/pull/66/commits/db32bf6ab80e2fae01bc3c5a367436d417453ee9)

**Review by domsteinbach — APPROVED** [review]
Thank you! All fine of course.

**Pull request merged**